src/snowflake/snowpark/mock/util.py

```diff
@@ -164,3 +187,33 @@ def process_string_time_with_fractional_seconds(time: str, fractional_seconds) -
         seconds_part = seconds_part[: min(idx, fractional_seconds)] + seconds_part[idx:]
         ret = f"{time_parts[0]}.{seconds_part}"
     return ret
+
+
+def fix_drift_between_column_sf_type_and_dtype(col: ColumnEmulator):
+    if (
+        isinstance(col.sf_type.datatype, _NumericType)
+        and col.apply(lambda x: x is None).any()
+    ):  # non-object dtype converts None to NaN for numeric columns
+        return col
+    sf_type_to_dtype = {
+        ArrayType: object,
+        BinaryType: object,
+        BooleanType: bool,
+        ByteType: numpy.int8 if not col.sf_type.nullable else "Int8",
+        DateType: object,
+        DecimalType: numpy.int64 if not col.sf_type.nullable else "Int64",
```

Comment:
Large decimals are over the max value of int64. It needs to be an object.
If you choose not to support now but better have a JIRA to follow-up.

Reply:
I created https://snowflakecomputing.atlassian.net/browse/SNOW-981994